bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -127,32 +165,65 @@ private AttributesImpl getLinkAttributes(String pagePath) {
         return attributes;
     }
 
-    @Override
-    public void setContentHandler(ContentHandler contentHandler) {
-        this.contentHandler = contentHandler;
-    }
+    /**
+     * Aggregates the AMP specific component js.
+     * @param request The request used to resolve the page and its resources.
+     * @return The aggregated AMP component js.
+     */
+    private String getJsContent(SlingHttpServletRequest request) {
 
-    @Override
-    public void dispose() { }
+        if (StringUtils.isBlank(cfg.getHeadlibName())) {
+            LOG.error("Headlib name not defined. Failed to aggregate AMP component js.");
+            return "";
+        }
 
-    @Override
-    public void setDocumentLocator(Locator locator) {
-        contentHandler.setDocumentLocator(locator);
+        StringBuilder output = new StringBuilder();
+
+        // Retrieve a set of the component's resource types.
+        Set<String> resourceTypes =
+            Utils.getResourceTypes(request.getResource(), cfg.getHeadlibResourceTypeRegex(), new HashSet<>());
+
+        // Iterate through each resource type and read its AMP headlib.
+        for (String resourceType : resourceTypes) {
+
+            // Resolve the resource type's AMP headlib.
+            Resource headLibResource;
+            String headLibPath = resourceType + "/" + cfg.getHeadlibName() + "/" + JcrConstants.JCR_CONTENT;
+            headLibResource = Utils.resolveResource(request.getResourceResolver(), headLibPath);
```

Comment:
won't this cause issues when the resource resolver from the request is an anonymous session? I would think you'd want to use the service user here since it's goign to be looking directly in /apps/ and /libs/